packages/stryker/src/transpiler/MutantTranspileScheduler.ts

```diff
@@ -0,0 +1,87 @@
+import { Observable, range } from 'rxjs';
+import TestableMutant from '../TestableMutant';
+import { File } from 'stryker-api/core';
+import SourceFile from '../SourceFile';
+import { Transpiler } from 'stryker-api/transpile';
+import { Disposable } from 'stryker-api/plugin';
+import TranspiledMutant from '../TranspiledMutant';
+import TranspileResult from './TranspileResult';
+import { errorToString } from '@stryker-mutator/util';
+import { tokens } from 'stryker-api/plugin';
+import { coreTokens } from '../di';
+import { from } from 'rxjs';
+import { flatMap, zip } from 'rxjs/operators';
+import { BehaviorSubject } from 'rxjs';
+
+const INITIAL_CONCURRENCY = 100;
+
+export class MutantTranspileScheduler {
+
+  private currentMutatedFile: SourceFile;
+  private readonly concurrencyTicket$ = new BehaviorSubject<number>(INITIAL_CONCURRENCY);
+
+  public static inject = tokens(coreTokens.transpiler, coreTokens.transpiledFiles);
+
+  /**
+   * Creates a mutant transpiler
+   */
+  constructor(private readonly transpiler: Transpiler & Disposable, private readonly unMutatedFiles: ReadonlyArray<File>) { }
+
+  public scheduleTranspileMutants(allMutants: ReadonlyArray<TestableMutant>): Observable<TranspiledMutant> {
+    return from(allMutants).pipe(
+      zip(this.concurrencyTicket$.pipe(
+        flatMap(n => range(0, n))
+      )),
+      flatMap(([mutant]) => this.transpileMutant(mutant), 1 /* IMPORTANT! Never transpile multiple mutants at once! */)
+    );
+  }
+
+  /**
+   * Schedule next mutant to be transpiled
+   */
+  public readonly scheduleNext = () => {
+    this.concurrencyTicket$.next(1);
+  }
+
+  /**
+   * Dispose all
+   */
+  public dispose() {
+    this.concurrencyTicket$.complete();
+    // TODO: Let typed-inject dispose this one
```

Comment:
Is this TODO still relevant?

Reply:
Yes it is. Once all disposable instances are created with `typed-inject` we can use it to dispose all our stuff. As for now, we cannot do that, so that's what this hack is for.